server/api/views.py

```diff
@@ -158,3 +158,19 @@ def get(self, request, *args, **kwargs):
             writer.writerow(row)
 
         return response
+
+class CategoryRemovalView(APIView):
+    def post(self, request):
+        category_id = request.data.get('id')
+        if not category_id:
+            return JsonResponse({'error': 'Missing category_id'}, status=status.HTTP_400_BAD_REQUEST)
+        
+        try:
+            category = Category.objects.get(pk=category_id)
+        except Category.DoesNotExist:
+            return JsonResponse({'error': 'Category not found'}, status=status.HTTP_404_NOT_FOUND)
+        
+        Transaction.objects.filter(category=category).update(category=None)
```

Comment:
`Transaction.objects` refers to a custom model manager in `managers.py`, which is responsible for merging, filtering and sorting transactions. This takes time and isn't really needed in this case, so it will probably be faster to use `Transaction.admin_objects` which is the default Django model manager and is much simplier.

However it's necessary to compare resulting SQL to be sure